platform/darwin/src/MGLPolygon.mm

```diff
@@ -54,6 +55,14 @@ - (NSUInteger)hash {
     return [super hash] + [[self geoJSONDictionary] hash];
 }
 
+- (CLLocationCoordinate2D)coordinate {
+    // pole of inaccessibility
+    auto poi = mapbox::polylabel([self polygon]);
+    CLLocationCoordinate2D centroid = MGLLocationCoordinate2DFromPoint(poi);
+    
+    return centroid;
```

Comment:
FYI, it's perfectly acceptable to return the result of MGLLocationCoordinate2DFromPoint() directly without creating a local variable for it. If you need to verify the return value here, Xcode's debugger displays a "Return Value" entry in the Variables view when stepping out of a function.